Firestore/Source/API/FIRFirestore.mm

```diff
@@ -80,6 +80,35 @@ @implementation FIRFirestore {
   return instances;
 }
 
++ (void)load {
+  NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
+  [center addObserverForName:kFIRAppDeleteNotification
+                      object:nil
+                       queue:nil
+                  usingBlock:^(NSNotification *_Nonnull note) {
+                    NSString *appName = note.userInfo[kFIRAppNameKey];
+                    if (appName == nil) return;
+
+                    NSMutableDictionary *instances = [self instances];
+                    @synchronized(instances) {
+                      // Since the key for instances isn't just the app name, iterate over all the
+                      // keys to get the one(s) we have to delete. There could be multiple in case
+                      // the user calls firestoreForApp:database:.
+                      NSMutableArray *keysToDelete = [[NSMutableArray alloc] init];
+                      for (NSString *key in instances.allKeys) {
+                        if ([key hasPrefix:appName]) {
```

Comment:
Please add a comment to the key construction code that notes that the app name must be the prefix for the key or deletion will break.

Also, this is broken if you have delete an app whose name is a prefix of another app name.

Reply:
Good idea, fixed the prefix issue too.